lib/spec/router.sw-spec.js

```diff
@@ -4,43 +4,186 @@ importScripts('/base/lib/router.js');
+    });
+  });
+
+  describe('match()', function() {
+    var defaultMiddleware = function() {};
+
+    it('should match a single middleware', function() {
+      router.get('/', defaultMiddleware);
+      expect(router.match('get', '/')).to.deep.equal([defaultMiddleware]);
+    });
+
+    it('should match multiple middlewares', function() {
+      var mw2 = function () {};
+      router.get('/', defaultMiddleware);
+      router.get('/', mw2);
+      expect(router.match('get', '/')).to.deep.equal([defaultMiddleware, mw2]);
+    });
+
+    it('should match a regular expression', function() {
+      router.get('/[0-9]+', defaultMiddleware);
+      expect(router.match('get', '/1')).to.deep.equal([defaultMiddleware]);
+    });
+
+    it('should prevent from matching a regular expression', function() {
+      router.get('/[0-9]+', defaultMiddleware);
+      expect(router.match('get', '/')).to.deep.equal([]);
+    });
+
+    it('should lowercase the method name', function() {
+      router.get('/', defaultMiddleware);
+      expect(router.match('GET', '/')).to.deep.equal([defaultMiddleware]);
+    });
+
+    it('should return an empty array if the method name is unknown', function() {
+      expect(router.match('invalidMethod', '/')).to.deep.equal([]);
+    });
+
+    it('should return an empty array if there is no middleware attached to this particular method', function() {
+      router.get('/', defaultMiddleware);
+      expect(router.match('post', '/')).to.deep.equal([]);
+    });
+
+    it('should return only one middleware even if multiple are attached to the same route (but not the same method)', function() {
+      var mw2 = function () {};
+      router.get('/', defaultMiddleware);
+      router.post('/', mw2);
+      expect(router.match('get', '/')).to.deep.equal([defaultMiddleware]);
+    });
+
+    it('should return only one middleware even if the same middleware was attached to multiple routes', function() {
+      router.get('/a', defaultMiddleware);
+      router.get('/b', defaultMiddleware);
+      expect(router.match('get', '/a')).to.deep.equal([defaultMiddleware]);
+    });
+
+    describe('"all" method', function() {
+      it('should match every single method', function() {
+        router.all('/', defaultMiddleware);
+        expect(router.match('get', '/')).to.deep.equal([defaultMiddleware]);
+        expect(router.match('post', '/')).to.deep.equal([defaultMiddleware]);
+        expect(router.match('put', '/')).to.deep.equal([defaultMiddleware]);
+        expect(router.match('delete', '/')).to.deep.equal([defaultMiddleware]);
+        expect(router.match('head', '/')).to.deep.equal([defaultMiddleware]);
+      });
+
+      it('should match "all"', function() {
+        router.all('/', defaultMiddleware);
+        expect(router.match('all', '/')).to.deep.equal([defaultMiddleware]);
+      });
+
+      // Currently failing, looks like a bug to me
+      it('should not match an invalid method', function() {
```

Comment:
It is consistent with the [test about matching an invalid method](https://github.com/arcturus/serviceworkerware/pull/15/files#diff-eb6b204faa5b086f4603f7bfe4fc3718R145)